test/integration/component/test_volumes.py

```diff
@@ -603,7 +603,134 @@ def test_01_volume_attach_detach(self):
                 "Check the state of VM"
             )
         except Exception as e:
-            self.fail("Exception occuered: %s" % e)
+            self.fail("Exception occurred: %s" % e)
+        return
+    
+    @attr(tags=["advanced", "advancedns"])
+    def test_02_root_volume_attach_detach(self):
+        """Test Root Volume attach/detach to VM
+        """
+
+        # Validate the following
+        # 1. Deploy a VM
+        # 2. Check for root volume
+        # 3. Stop VM
+        # 4. Detach root volume
+        # 5. Verify root volume detached
+        # 6. Attach root volume
+        # 7. Start VM
+        
+        try:
+            # Check for root volume
```

Comment:
First Verify the hypervisor type is KVM  in order for this test to execute else skip this test . As root volume detach is supported for KVM .